### ISSUE TITLE
508-defect-1 ❗️ Launchblocker [SCREENREADER]: Focus must be managed for single-page applications

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#22292]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I want to know where I am within the form at all times.

## Details

Since focus is currently not managed within the form, screen reader users may become lost due to:
- title tags not being unique or descriptive of steps in the form
- h1s not being unique or descriptive of steps in the form

## Acceptance Criteria

- [ ] Focus is managed so screen readers know _what_ they are completing on each page of the form
  - [ ] Introduction page
  - [ ] Form field pages
  - [ ] Results page

## Steps to Recreate

1. Enter `https://staging.va.gov/health-care/covid-19-vaccine/stay-informed` in browser
2. Start your choice of screenreader
3. Navigate to the sign up now button and hit enter
4. Confirm focus is not managed (goes to default "Home" in breadcrumb)

## Proposed Solution

Ideally, both page titles and `h1`s should be unique on every page. Focus within the form flow should automatically be placed on the `h1`.

If this isn't possible because of the form system, I recommend following the pattern that exists on the rest of VA.gov by:
- Adding a unique `h2` to each page (except for the results page)
- Upon pressing "continue" or starting the flow with "Sign up now," focus should be placed on the unique `h2`, not the generic `h1` "Sign up to get a COVID-19 vaccine at VA"

### Solution A
<img width="967" alt="Screen Shot 2021-03-29 at 11 31 23 AM" src="https://user-images.githubusercontent.com/14154792/112861167-8ebf9300-9082-11eb-841c-5b8bd2edb14f.png">

### Solution B
<img width="821" alt="Screen Shot 2021-03-29 at 11 29 46 AM" src="https://user-images.githubusercontent.com/14154792/112861145-8a937580-9082-11eb-8fbd-16d4402dd195.png">

### Lack of focus management
Without focus management within the form, users must either use the skip to content link or constantly tab through everything in order to figure out what the next step of the form is about

https://user-images.githubusercontent.com/14154792/112863043-6d5fa680-9084-11eb-86a8-e59342b5e5ad.mov

**Comment by cohnjesse** [comment]
@joshkimux I totally agree that this needs to be fixed however this is a form system issue and therefore should not be a launch blocker for anyone. This should be fixed at the platform level and should not block anyone from launching before it is fixed there, fixing it at the app level only creates further complications in the codebase that will need to be cleaned up by the platform team later. @batemapf @artsymartha68 for visibility.

**Comment by joshkimux** [comment]
@1Copenut for awareness on focus related tickets.

@cohnjesse To clarify, is this in response to solution A or B that I've documented in the ticket? Solution B was written in reflection on how the form system currently manages focus (consistent h1, but focus is sent to the h2 which is unique to each page). 

Would we be able to find an alternative way of closing out the ticket that is in the best interest of all parties?

**Comment by stephenBarrs** [comment]
@joshkimux @artsymartha68 we could add the h2 and force focus on it when the page loads.  Does that have to be a new h2 or could we focus on `To get started, tell us about your experience with VA health care. We'll then guide you to the right form.` which is currenlty in a `<span>`

**Comment by joshkimux** [comment]
@stephenBarrs If we want to be consistent with the form system, focus should be set on that new `h2` rather than the `p` below. It might be a little odd because usually the `h2`s are steppers e.g. "Step 1 of X, Determine the right form."

Might be worth wrapping folks with experience with the form system + content in here

**Comment by stephenBarrs** [comment]
@artsymartha68 how do you feel about adding an h2 to the intro screen like in `Solution B` above?

**Comment by artsymartha68** [comment]
Can we ask @DanielleThierryUSDSVA too? I would like to do what makes sense for the content as well as a11y if it's possible.

**Comment by joshkimux** [comment]
@stephenBarrs Also curious to hear what the lift would be to simply change the `h1`s for each of these pages so that they are unique. I think that would be a more straightforward path of fixing it if this isn't actually form system dependent.

**Comment by artsymartha68** [comment]
It's my understanding that the default form system behavior is the same H1 throughout all screens. Another opportunity to make a global change to benefit everyone!

**Comment by stephenBarrs** [comment]
The h1s on each page are controlled by forms system.

**Comment by joshkimux** [comment]
Roger that. Focus management towards unique `h2`s would be the right way forwards!

**Comment by caw310** [comment]
We're not sure what you did to resolve but see that you now have unique H1s but not H2s. 
Were you able to fix this or address this without a form system change?

**Comment by SarahKay8** [comment]
Focus has been forced on unique H2s per Josh's recommendations.https://images.zenhubusercontent.com/133843125/226d7137-66c4-45a1-9f53-e0910c3d4be0/img_6832__1_.mov

**Comment by SarahKay8** [comment]
Focus is forced on unique H2s per Josh's recommendations. Closing out. ![Image 8-19-22 at 12.23 PM.jpg](https://images.zenhubusercontent.com/623cd4d6774110eeb19922cb/f3fe0389-69c3-4886-aad5-f587bb4fbec8)
